File: python/calculate-sum-squares.py
import threading

threads = []
results = [0 for _ in range(5)]

def sum_of_squares(num):
    re = 0
    i = 1
    while i <= num:
        re = re + (i*i)
        i = i + 1
    return re

def run(start, end, index):
    i = start
    result = 0
    while i <= end:
        result = result + sum_of_squares(i)
        i = i + 1
    results[index] = result
    print(f"Sum of squares from {start} to {end} is {result}")

if __name__ == "__main__":
    ranges = [
        (100, 200),
        (201, 300),
        (301, 400),
        (401, 500),
        (501, 600),
    ]
    i = 0
    while i < 5:
        t = threading.Thread(target=run(ranges[i][0],  ranges[i][1], i), args=(i,))
        threads.append(t)
        t.start()
        i = i + 1
    i = 0
    while i < 5:
        threads[i].join()
        i = i + 1
    total = 0
    i = 0
    while i < 5:
        total = total + results[i]
        i = i + 1
    print(f"Total Sum is: {total}")

File: go/calculate-sum-squares.go
package main

import (
  "fmt"
  "sync"
)

type ThreadValues struct {
  start int
  end int
  result int64
}

func sum_of_squares(num int) int64 {
  var re int64
  for i := 0; i <= num; i++ {
    re += int64(i * i)
  }
  return re
}

func (tv *ThreadValues) run(wg *sync.WaitGroup) {
  defer wg.Done()
  for i := tv.start; i <= tv.end; i++ {
    tv.result += sum_of_squares(i)
  }
  fmt.Printf("Sum of squares from %d to %d is %d!\n", tv.start, tv.end, tv.result)
}

func (tv *ThreadValues) get_result() int64 {
  return tv.result
}

func main() {
  ranges := [][]int{
    {100, 200},
    {201, 300},
    {301, 400},
    {401, 500},
    {501, 600},
  }

  var threads []*ThreadValues
  var wg sync.WaitGroup

  for _, r := range ranges {
    thread := &ThreadValues{start: r[0], end: r[1], result: 0}
    threads = append(threads, thread)
    wg.Add(1)
    go thread.run(&wg)
  }

  wg.Wait()

  var total int64
  for _, tv := range threads {
    total += tv.get_result()
  }
  fmt.Printf("Total Sum is: %d\n", total)
}
